boards/nxp/mcx_nxxx_evk/Kconfig

```
@@ -2,4 +2,9 @@
 # SPDX-License-Identifier: Apache-2.0
 
 config BOARD_MCX_N9XX_EVK
+	bool
+	select BOARD_EARLY_INIT_HOOK
+
+config BOARD_MCX_N5XX_EVK
+	bool
```

```suggestion
config BOARD_NEEDS_EARLY_HOOK
        bool
        default y if BOARD_MCX_N9XX_EVK || BOARD_MCX_N5XX_EVK
        select BOARD_EARLY_INIT_HOOK
```
or alternatively filter on e.g soc family, soc series, etc.